### PR TITLE
feat: Auto refresh jobs and events.

### PR DESCRIPTION
**Pull request by OronNadiv**
## Context

In both workflow and list views, the user should be able to:
1. See new events/jobs as they triggered on other tabs/computers.
1. See a job starts/stops without clicking the "refresh" button.

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by jithine — APPROVED** [review]

File: app/components/pipeline-list-view/component.js [hunk 1]
```diff
@@ -208,6 +209,16 @@ export default Component.extend({
 
     return rows;
   },
+  jobsObserver: observer('jobsDetails.[]', function jobsObserverFunc({ jobsDetails }) {
+    const rows = this.getRows(jobsDetails);
+    const lastRows = get(this, 'lastRows') || [];
+    const isEqualRes = isEqual(rows.map(r => r.job), lastRows.map(r => r.job));
```
Comment (adong) [inline app/components/pipeline-list-view/component.js:215]: I'm not sure if the response returns jobs in a consistent order, maybe we need to `sort` before comparing.

Comment (adong) [reply app/components/pipeline-list-view/component.js:215]: I could be wrong, but I don't think the observer's function has parameters.

```
jobsObserverFunc({ jobsDetails }) {
    const rows = this.getRows(jobsDetails);
```

So this function always sets the table with new rows.

File: app/pipeline/jobs/index/controller.js [hunk 2]
```diff
@@ -1,14 +1,39 @@
...
+import { get, set, computed } from '@ember/object';
 import { jwt_decode as decoder } from 'ember-cli-jwt-decode';
 
 import ENV from 'screwdriver-ui/config/environment';
 import ModelReloaderMixin from 'screwdriver-ui/mixins/model-reloader';
-import { isPRJob } from 'screwdriver-ui/utils/build';
+import { isPRJob, isActiveBuild } from 'screwdriver-ui/utils/build';
+import moment from 'moment';
 import { createEvent, startDetachedBuild, stopBuild, updateEvents } from '../../events/controller';
+import { SHOULD_RELOAD_SKIP, SHOULD_RELOAD_YES } from '../../../mixins/model-reloader';
```
Comment (adong) [inline app/pipeline/jobs/index/controller.js:11]: Ember know the relative path pain, and you could change the syntax from 

`'../../../mixins/model-reloader';`

to 

`'screwdriver-ui/mixins/model-reloader';`

File: app/pipeline/jobs/index/controller.js [hunk 3]
```diff
@@ -22,11 +47,14 @@ export default Controller.extend(ModelReloaderMixin, {
     });
   },
 
-  reload() {
+  async reload() {
     try {
       this.send('refreshModel');
+      await this.refreshListViewJobs();
```
Comment (adong) [inline app/pipeline/jobs/index/controller.js:53]: I dont think this guarantees `model()` is fetched and loaded when line `this.refreshListViewJobs()` runs.

File: app/pipeline/jobs/index/controller.js [hunk 4]
```diff
@@ -192,8 +220,14 @@ export default Controller.extend(ModelReloaderMixin, {
       }
 
       await this.createEvent(eventPayload);
+      await this.reload();
+      set(this, 'lastRefreshed', moment().subtract(1, 'day'));
```
Comment (adong) [inline app/pipeline/jobs/index/controller.js:224]: `moment().subtract(1, 'day')`, I don't get the logic here, why the date is set to yesterday?

**Review by adong — CHANGES_REQUESTED** [review]
👀

File: app/pipeline/jobs/index/controller.js (hunk 2 above)
Comment (OronNadiv) [reply app/pipeline/jobs/index/controller.js:11]: 👍

File: app/pipeline/jobs/index/controller.js (hunk 4 above)
Comment (OronNadiv) [reply app/pipeline/jobs/index/controller.js:224]: I wanted to reset the "lastRefreshed" value to trigger another refresh.

File: app/components/pipeline-list-view/component.js (hunk 1 above)
Comment (OronNadiv) [reply app/components/pipeline-list-view/component.js:215]: Done. 👍

File: app/pipeline/jobs/index/controller.js (hunk 3 above)
Comment (OronNadiv) [reply app/pipeline/jobs/index/controller.js:53]: Good point but I can't find a use-case where refreshListViewJobs does not refresh the list.

File: app/components/pipeline-list-view/component.js [hunk 5]
```diff
@@ -208,6 +209,19 @@ export default Component.extend({
 
     return rows;
   },
+  jobsObserver: observer('jobsDetails.[]', function jobsObserverFunc({ jobsDetails }) {
+    const rows = this.getRows(jobsDetails);
```
Comment (adong) [inline app/components/pipeline-list-view/component.js:213]: I have to say that I didn't think there would be parameters for observer function,  and I was wrong.

It indeed has
```
observer('value', function(sender, key, value, rev) {
```

see Ref: https://api.emberjs.com/ember/3.20/classes/Observable/methods

File: app/components/pipeline-list-view/component.js [hunk 6]
```diff
@@ -208,6 +209,19 @@ export default Component.extend({
...
+    const lastRows = get(this, 'lastRows') || [];
+    const isEqualRes = isEqual(
+      rows.map(r => r.job).sort((a, b) => (a.jobName || '').localeCompare(b.jobName)),
+      lastRows.map(r => r.job).sort((a, b) => (a.jobName || '').localeCompare(b.jobName))
+    );
```
Comment (adong) [inline app/components/pipeline-list-view/component.js:218]: 👍

File: app/pipeline/jobs/index/controller.js (hunk 3 above)
Comment (adong) [reply app/pipeline/jobs/index/controller.js:57]: It's okay.

**Review by adong — APPROVED** [review]
Looks good to me 👍

**Review by jithine — APPROVED** [review]

**Pull request merged**